ipn/ipnauth/nonwin.go

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package ipnauth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"runtime"
+	"strconv"
+	"time"
+
+	"github.com/tailscale/peercred"
+	"tailscale.com/envknob"
+	"tailscale.com/ipn"
+	"tailscale.com/safesocket"
+	"tailscale.com/types/logger"
+	"tailscale.com/util/groupmember"
+	"tailscale.com/util/osuser"
+	"tailscale.com/version"
+	"tailscale.com/version/distro"
+)
+
+var (
+	errMustBeRootOrOperator       = ipn.NewAccessDeniedError("must be root or an operator")
+	errMustBeRootOrSudoerOperator = ipn.NewAccessDeniedError("must be root, or be an operator and able to run 'sudo tailscale' to serve a path")
+)
+
+// NonWindowsIdentity is a non-Windows user identity.
+type NonWindowsIdentity struct {
+	goos  string
+	creds *peercred.Creds // or nil
+
+	// tests are fields used exclusively for testing purposes.
+	tests struct {
+		uid, username *string
+		isAdmin       *bool
+	}
+}
+
+// NewNonWindowsIdentity returns a new identity representing a user identity on
+// the specified goos and with optional peercred.Creds.
+// It panics if goos is "windows".
+func NewNonWindowsIdentity(goos string, creds *peercred.Creds) (*NonWindowsIdentity, error) {
+	if goos == "windows" {
+		return nil, errors.New("goos must not be `windows`")
+	}
+	return &NonWindowsIdentity{goos: goos, creds: creds}, nil
+}
+
+// UserID returns "".
```

Comment:
can delete this

or say // UserID returns the empty string; it exists only to implement Foo.